src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+  //
+  for (ix = 0; ix < scrP->attributeList.attributeV.size(); ++ix)
+  {
+    attributes.push_back(scrP->attributeList.attributeV[ix]);
+  }
+
+
+  //
+  // 4. throttling
+  //
+  throttling = scrP->throttling.parse();
+
+
+  //
+  // 5. expirationTime
+  //
+  expirationTime = _expiration;
+
+
+  //
+  // 6. restriction
+  //
+  restriction.fill(&scrP->restriction);
+
+
+  //
+  // 7. notifyConditionVector
+  //
+  notifyConditionVector.fill(scrP->notifyConditionVector);
+  LM_M(("Added %d notifyConditions", notifyConditionVector.size()));
+
+  //
+  // 8. reference
+  //
+  reference.set(scrP->reference.get());
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::Subscription - 
+*/
+Subscription::Subscription
+(
+  const std::string&               _tenant,
+  const std::string&               _servicePath,
+  const std::string&               _subscriptionId,
+  const std::vector<EntityInfo*>&  _entityIdInfos,
+  const std::vector<std::string>&  _attributes,
+  int64_t                          _throttling,
+  int64_t                          _expirationTime,
+  const Restriction&               _restriction,
+  NotifyConditionVector&           _notifyConditionVector,
+  const std::string&               _reference,
+  Format                           _format
+)
+{
+  unsigned int ix;
+
+  tenant         = _tenant;
+  servicePath    = _servicePath;
+  subscriptionId = _subscriptionId;
+  format         = _format;
+
+  for (ix = 0; ix < _entityIdInfos.size(); ++ix)
+  {
+    entityIdInfos.push_back(_entityIdInfos[ix]);
+  }
+
+  for (ix = 0; ix < _attributes.size(); ++ix)
+  {
+    attributes.push_back(_attributes[ix]);
+  }
+
+  throttling     = _throttling;
+  expirationTime = _expirationTime;
+
+  restriction.fill((Restriction*) &_restriction);
+  notifyConditionVector.fill(_notifyConditionVector);
+  reference.set(_reference);
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::entityIdInfoAdd - 
+*
+* FIXME P10: need to decide whether this function copies the EntityInfo or
+*            just uses the pointer. For now, just using the pointer.
+*/
+void Subscription::entityIdInfoAdd(EntityInfo* entityIdInfoP)
+{
+  entityIdInfos.push_back(entityIdInfoP);
+}
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::release - 
+*/
+void EntityInfo::release(void)
+{
+  regfree(&entityIdPattern);
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::attributeMatch - 
+*/
+bool Subscription::attributeMatch(const std::string& attributeName)
+{
+  LM_M(("KZ: attributeMatch: %d notifyConditions", notifyConditionVector.size()));
+
+  for (unsigned int ncvIx = 0; ncvIx < notifyConditionVector.size(); ++ncvIx)
+  {
+    NotifyCondition* ncP = notifyConditionVector[ncvIx];
+
+    if (strcasecmp(ncP->type.c_str(), "ONCHANGE") != 0)
```

Comment:
My idea was to have this function more "generic", but it was a bad idea from the start.
Removing the check - let's not waste any time ...